core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabase.java

```diff
@@ -117,6 +117,11 @@ public Scanner scan(Scan scan) throws ExecutionException {
       close(connection);
       throw new ExecutionException(
           CoreError.JDBC_ERROR_OCCURRED_IN_SELECTION.buildMessage(e.getMessage()), e);
+    } catch (Exception e) {
+      if (connection != null) {
```

Comment:
Don't we need to `connection.rollback()` here?

Reply:
Thanks! I focused on the connection leak, but I think the transaction should be rolled-back immediately to avoid potential future lazy-recovery. Addressed it in https://github.com/scalar-labs/scalardb/pull/2766/commits/e96888e673ea7f9aea35c6622e4937a2b448bf61.